daemon/create.go

```diff
@@ -150,6 +150,14 @@ func (daemon *Daemon) generateSecurityOpt(ipcMode containertypes.IpcMode, pidMod
 
 		return label.DupSecOpt(c.ProcessLabel), nil
 	}
+	if pidContainer := pidMode.Container(); pidContainer != "" {
+		c, err := daemon.GetContainer(pidContainer)
+		if err != nil {
+			return nil, err
+		}
+
+		return label.DupSecOpt(c.ProcessLabel), nil
```

Comment:
We can't join two namespaces of two different containers unless transitively they have the same label.
Cc: @rhatdan

Sent from my iPhone

> On May 11, 2016, at 8:32 PM, Kenfe-Mickaël Laventure <notifications@github.com> wrote:
> 
> In daemon/create.go:
> 
> > @@ -150,6 +150,14 @@ func (daemon *Daemon) generateSecurityOpt(ipcMode containertypes.IpcMode, pidMod
> >  
> >  		return label.DupSecOpt(c.ProcessLabel), nil
> >  	}
> > +	if pidContainer := pidMode.Container(); pidContainer != "" {
> > +		c, err := daemon.GetContainer(pidContainer)
> > +		if err != nil {
> > +			return nil, err
> > +		}
> > +
> > +		return label.DupSecOpt(c.ProcessLabel), nil
> If we have both ipcMode.IsContainer() and pidMode.IsContainer() are true and they both point to different containers with different labels we will never apply the second container labels.
> 
> I'm no selinux expert, but shouldn't those be merged somehow?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
>